src/EditorFeatures/CSharpTest/QualifyMemberAccess/QualifyMemberAccessTests.cs

```diff
@@ -1143,5 +1143,23 @@ static Program()
 }",
 CodeStyleOptions.QualifyEventAccess);
         }
+
+        [WorkItem(21519, "https://github.com/dotnet/roslyn/issues/21519")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsQualifyMemberAccess)]
+        public async Task QualifyPropertyAccess_NameofArgument()
+        {
+            await TestAsyncWithOption(
+@"class Program
+{
+    public int Foo { get; set; }
+    public string Bar = nameof([|Foo|]);
+}",
+@"class Program
+{
+    public int Foo { get; set; }
+    public string Bar = nameof(this.Foo);
```

Comment:
@sharwell I've fixed it, but I found another issue.
I this case:
```csharp
public class C
    {
        public int Foo { get; set; }
        public string Bar { get => nameof(Foo); }
    }
```
```CSharpQualifyMemberAccessDiagnosticAnalyzer``` doesn't even analyze it (so there isn't IDE0009 suggestion here).
Should it be fixed in this PR or I should file a separate issue?